### PR TITLE
[release-1.0] Add Close() handling to kubernetesenv adapter

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR is the `release-1.0` version of PR  #10839. It is meant to address #10393.

### CODE REVIEW

File: mixer/adapter/kubernetesenv/kubernetesenv.go [hunk 1]
```diff
@@ -183,7 +185,13 @@ func (h *handler) GenerateKubernetesAttributes(ctx context.Context, inst *ktmpl.
 	return out, nil
 }
 
-func (h *handler) Close() error {
+func (h *handler) Close() (err error) {
+	defer func() {
+		if recover() != nil {
```
Comment (hklai) [inline mixer/adapter/kubernetesenv/kubernetesenv.go]: sorry i am not familiar with this code. Is recover() defined somewhere?

Comment (ozevren) [reply mixer/adapter/kubernetesenv/kubernetesenv.go]: It is part of golang: https://blog.golang.org/defer-panic-and-recover

**Comment by douglas-reid** [comment]
@mandarjog friendly ping. this came up in the `release-1.1` tracking meeting.

**Comment by hklai** [comment]
Ignore the linter errors. It has been broken in release-1.0 for a while.

**Comment by wenchenglu** [comment]
@hklai 
not just lint, test also failed. I remember only lint failed not test.

/lgtm
/approve

**Comment by mandarjog** [comment]
@wattli can you check why test is failing?
```
Your job ran 4888 tests with 1 failure
TestGcpGetServiceIdentity - probe
Failed
controller_test.go:37: cannot verify the cert with the provided root chain and cert pool
panic: runtime error: invalid memory address or nil pointer dereference
/usr/local/go/src/testing/testing.go:742 +0x29d
/usr/local/go/src/runtime/panic.go:502 +0x229
/go/src/istio.io/istio/security/pkg/pki/util/keycertbundle.go:135 +0x37
/go/src/istio.io/istio/security/pkg/pki/ca/ca.go:174 +0x4c
/go/src/istio.io/istio/security/pkg/probe/controller.go:93 +0x151
/go/src/istio.io/istio/security/pkg/probe/controller_test.go:92 +0x781
/usr/local/go/src/testing/testing.go:777 +0xd0
/usr/local/go/src/testing/testing.go:824 +0x2e0
```

**Comment by douglas-reid** [comment]
fwiw, that unit test is completely unrelated to this code change... and passed on prow. it appears to be a flake in the CI.

**Comment by douglas-reid** [comment]
@wattli that test seems to fail repeatedly?  we really need this in so that we can cut 1.0.6.

**Comment by wattli** [comment]
Looking, and will give an update soon

**Comment by wattli** [comment]
It's weird since other PRs (https://github.com/istio/istio/commits/release-1.1) passed the test with no problem, and the prow test passes. I just re-ran the test and let's see how it goes. 

@myidpt did have a PR modifying this test in early Jan. But I don't think it should be related...

**Comment by douglas-reid** [comment]
@wattli I agree that it is strange. The prow test passing seems to indicate this is a CI issue (to me at least).

**Comment by wattli** [comment]
@douglas-reid , I suspect it's related to the lint error, 

"please run bin/fmt.sh against: ./security/pkg/registry/kube/serviceaccount.go
Makefile:266: recipe for target 'lint' failed"

Can we address that first? And did you sync to the latest version?

**Comment by wattli** [comment]
I tested it manually that the cert is still valid and the verification passed. And that's the case for the recent PRs merged in this branch (https://github.com/istio/istio/pull/11278, https://github.com/istio/istio/pull/11388, etc). 

I would rebase and see how it goes...

**Comment by wattli** [comment]
Just created a new PR to add more log to that error, and let's see whether we can pass the test https://github.com/istio/istio/pull/11462

**Comment by wattli** [comment]
The new PR passed the circleci:test, please rebase and try again

**Comment by douglas-reid** [comment]
OK. rebased and force pushed.

**Comment by wattli** [comment]
And it (circleci: test) passed :)

**Comment by douglas-reid** [comment]
@wenchenglu would you mind using your powers for good?  appreciate all the help so far.

**Comment by hklai** [comment]
@mandarjog @douglas-reid ok to cut 1.0.6 now for testing (instead of kicking off a daily)?

We can retract in case it is bad.

**Comment by douglas-reid** [comment]
I'm not clear on the process of retraction... but it'd be nice to have a build to validate with the kNative folks in some manner.

**Pull request merged**